go/vt/wrangler/reparent.go

```diff
@@ -599,11 +607,34 @@ func (wr *Wrangler) plannedReparentShardLocked(ctx context.Context, ev *events.R
 		wr.logger.Infof("promote replica %v", masterElectTabletAliasStr)
 		event.DispatchUpdate(ev, "promoting replica")
 
+		waitCtx, waitCancel := context.WithTimeout(ctx, waitReplicasTimeout)
+		defer waitCancel()
+
+		err = wr.tmc.WaitForPosition(waitCtx, masterElectTabletInfo.Tablet, rp)
+
+		if err != nil || ctx.Err() != nil && ctx.Err() == context.DeadlineExceeded {
```

Comment:
This can be simplified, right?

```suggestion
		if err != nil || ctx.Err() == context.DeadlineExceeded {
```

Reply:
haha, yes. Don't know what I was thinking..